app/src/main/cpp/SJCL.cpp

```diff
@@ -176,7 +273,96 @@ char* SJCL::decrypt(string sjcl_json, string key) {
     free(cyphertext);
     free(data);
     free(derived_key);
-//    free(food);
+
+    return ret;
+}
+
+char* SJCL::encrypt(char* plaintext, const string& key) {
+    int plaintext_len = strlen(plaintext);
+    int iv_len = 13;    // use 13 because 15-lol (with initial lol=2) is hardcoded in the decryptccm implementation
+
+    // strange iv_len calculation due to the decryptccm implementation
+    if (plaintext_len >= 1<<16) iv_len--;
+    if (plaintext_len >= 1<<24) iv_len--;
+
+    int salt_len = 12;  //can I use a random number here?
+    int iter = 1000;
+    int key_size = 256;
+    int tag_size = 64;
+    int ciphertext_allocation_multiplicator = 3;    // give generated ciphertext some backup space
+
+    int ks = key_size / 8;  // Make it bytes
+    int ts = tag_size / 8;  // Make it bytes
+    unsigned char *ciphertext;
+    unsigned char *derived_key;
+    unsigned char tag[ts];
+    unsigned char *iv = nullptr;
+    unsigned char *salt = nullptr;
+    unsigned char *additional = (unsigned char *)"";
+    char* ret = NULL;
+
+    srand(time(NULL));
+
+    //iv = (unsigned char *) get_random_string(iv_len).c_str();
+    std::string tmpiv = get_random_string(iv_len);
+    iv = (unsigned char *) tmpiv.c_str();
```

Comment:
The IV does not need to be an string, it's preferred to be raw random bytes so constricting it to character bases would lower it's security.

Reply:
Fixed with the upcoming commit 1b96e278005c2c3fcffc483e53e5fea3f97e1cc8.